model/flow/execution_receipt.go

```diff
@@ -11,17 +12,36 @@ type Spock []byte
 // ExecutionReceipt is the full execution receipt, as sent by the Execution Node.
 // Specifically, it contains the detailed execution result. The `ExecutorSignature`
 // signs the `UnsignedExecutionReceipt`.
+//
+//structwrite:immutable - mutations allowed only within the constructor
 type ExecutionReceipt struct {
 	UnsignedExecutionReceipt
 	ExecutorSignature crypto.Signature
 }
 
-// UnsignedExecutionReceipt represents the unsigned execution receipt, whose contents the
-// Execution Node testifies to be correct by its signature.
-type UnsignedExecutionReceipt struct {
-	ExecutorID Identifier
-	ExecutionResult
-	Spocks []crypto.Signature
+// UntrustedExecutionReceipt is an untrusted input-only representation of a ExecutionReceipt,
+// used for construction.
+//
+// This type exists to ensure that constructor functions are invoked explicitly
+// with named fields, which improves clarity and reduces the risk of incorrect field
+// ordering during construction.
+//
+// An instance of UntrustedExecutionReceipt should be validated and converted into
+// a trusted ExecutionReceipt using NewExecutionReceipt constructor.
+type UntrustedExecutionReceipt ExecutionReceipt
+
+// NewExecutionReceipt creates a new instance of ExecutionReceipt.
+// Construction ExecutionReceipt allowed only within the constructor.
+//
+// All errors indicate a valid ExecutionReceipt cannot be constructed from the input.
+func NewExecutionReceipt(untrusted UntrustedExecutionReceipt) (*ExecutionReceipt, error) {
+	if len(untrusted.ExecutorSignature) == 0 {
+		return nil, fmt.Errorf("executor signature must not be empty")
+	}
+	return &ExecutionReceipt{
```

Comment:
we don't check validity of `untrusted.UnsignedExecutionReceipt`, I think we need to rely on respective constructor for this.